multipatch_analysis/ui/acq4_modules.py

```diff
@@ -61,6 +79,191 @@ def window(self):
+        except KeyError:
+            raise Exception('No Slice Selected')
+        
+        #check the image file path
+        safe_aff_image_path = lims.lims.safe_system_path(aff_image_path)
+
+        if os.path.exists(safe_aff_image_path) == False:
+            raise Exception("Couldn't find image in LIMS. Check you have the selected the correct slice.")
+        
+        aff_image_name = os.path.split(aff_image_path)[-1]
+        
+        #jpg_image_name = os.path.splitext(aff_image_name)[0] + ".jpg"
+
+        #image_service_url = "http://lims2/cgi-bin/imageservice?path="
+        #self.zoom = 6        #need to check that this ACQ4 can support this size
+
+        #full_url = image_service_url + aff_image_path + "&zoom={}".format(str(self.zoom))
+        save_path = os.path.join(self.base_path, aff_image_name)
+        safe_save_path = lims.lims.safe_system_path(save_path)
+
+        if os.path.exists(safe_save_path) == False:
+            with pg.BusyCursor():
+                shutil.copy2(safe_aff_image_path,safe_save_path)
+
+                # image = urllib.URLopener()
+                # image.retrieve(full_url, safe_save_path)
+                self.base_dir.indexFile(aff_image_name)
+                print("20x image moved")
+        self.image_20 = safe_save_path
+                
+
+    def create_lims_json(self):
+        """
+        Checks that markers are loaded into the canvas
+        Gets timestamp
+        Creates json for each cell clusters and saves locally
+        copies json to correct incoming folder
+        creates trigger file and drops in correct incoming folder
+        """
+        
+        items = self.mosaic_editor.canvas.items 
+        markers = [i for i in items if isinstance(i, MarkersCanvasItem)]
+        if len(markers) != 1:
+            raise Exception("Must have exactly 1 Markers item in the canvas.")
+        
+        for i in items:
+                try:
+                    if i.opts['handle'].name() == self.image_20:
+                        image = i
+                except AttributeError:
+                    pass
+
+        sites = set()
+        for item in items:
+            handle = item.opts.get('handle')
+            if handle is None:
+                continue
+            parent = handle.parent()
+            if parent.shortName().startswith('site_'):
+                sites.add(parent)
+
+        if len(sites) == 0:
+            raise Exception("No files loaded from site; can't determine which site to submit.")
+        if len(sites) > 1:
+            raise Exception("Files loaded from multiple sites; can't determine which site to submit.")
+        self.site_dh = list(sites)[0]
+        ts = self.site_dh.info()['__timestamp__']
+
+        
+        cluster_id = []
+        clusters = lims.cell_cluster_ids(self.slice_id)
+        for cid in clusters:
+            cluster_tmsp = lims.specimen_metadata(cid)['acq_timestamp']
```

Comment:
cluster_meta = lims.specimen_metadata(cid)
            if cluster_meta is None:
                continue
            cluster_tmsp = cluster_meta.get('acq_timestamp')